src/snowflake/cli/plugins/nativeapp/commands.py

```diff
@@ -69,6 +80,49 @@ def app_init(
     )
 
 
+@app.command("list-templates", hidden=True)
+def app_list_templates(**options) -> CommandResult:
+    """
+    Prints information regarding templates that can be used with snow app init.
+    """
+    with SecurePath.temporary_directory() as temp_path:
+        from git import Repo
+
+        repo = Repo.clone_from(
+            url=OFFICIAL_TEMPLATES_GITHUB_URL,
+            to_path=temp_path.path,
+            filter=["tree:0"],
+            depth=1,
+        )
+
+        # Close repo to avoid issues with permissions on Windows
+        repo.close()
+
+        # Mark a directory as a template if a project definition jinja template is inside
+        template_directories = [
+            entry.name
+            for entry in repo.head.commit.tree
+            if (temp_path / entry.name / "snowflake.yml.jinja").exists()
+        ]
+
+        # get the template descriptions from the README.md in its directory
+        template_descriptions = [
+            get_first_paragraph_from_markdown_file(
+                (temp_path / directory / "README.md").path
```

Comment:
Right now, this will fail if any templates are created that don't have top-level `README.md` files. I suggest we provide a fallback to `None` both in this case and when there is no paragraph text in the readme

Reply:
Good callout - added a test for it as well, thanks!